plugins/kubernetes/app/models/concerns/kubernetes/deploy_yaml.rb

```diff
@@ -30,10 +34,43 @@ def raw_template
       @raw_template ||= build.file_from_repo(kubernetes_role.config_file)
     end
 
-    def add_labels
-      labels.each do |key, value|
+    # This key replaces the default kubernetes key: 'deployment.kubernetes.io/podTemplateHash'
+    # This label is used by kubernetes to identify a RC and corresponding Pods
+    def set_rc_unique_label_key
+      template.spec.uniqueLabelKey = CUSTOM_UNIQUE_LABEL_KEY
```

Comment:
He He this is confusing.

So, that `CUSTOM_UNIQUE_LABEL_KEY`, which I have set to `rc_unique_identifier`, is just the name of the key that is going to be used to store the RC unique identifier which, by default, is `deployment.kubernetes.io/podTemplateHash`. But the identifier itself will be unique and auto-generated in the cluster. We don't have control over that.

The reason why I changed this key name was to make it more meaningful, when looking at the cluster data but also when using it on the code.